roles/provision_vm/tasks/main.yml

```diff
@@ -0,0 +1,56 @@
+---
+- name: Manage VM
+  community.vmware.vmware_guest:
+    hostname: "{{ provision_vm_hostname }}"
+    username: "{{ provision_vm_username }}"
+    password: "{{ provision_vm_password }}"
+    validate_certs: "{{ provision_vm_validate_certs | d(omit) }}"
+
+    port: "{{ provision_vm_port | d(omit) }}"
+    proxy_host: "{{ provision_vm_proxy_host | d(omit) }}"
+    proxy_port: "{{ provision_vm_proxy_port | d(omit) }}"
+
+    name: "{{ provision_vm_name }}"
```

Comment:
I think you can put `{{ provision_vm_name | mandatory }}` here, because it's required in all scenarios.